### PR TITLE
MSI magento-engcom/magento#65:

### PR DESCRIPTION
**Pull request by larsroettig**

### CODE REVIEW

File: app/code/Magento/InventoryApi/Test/_files/source_items_out_stock.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?php
```
Comment (naydav) [inline app/code/Magento/InventoryApi/Test/_files/source_items_out_stock.php]: Need to merge develop brunch and adapt tests according to possibility to pre-define id

File: app/code/Magento/Inventory/Indexer/StockItem/IndexDataProvider.php [hunk 2]
```diff
@@ -37,9 +39,18 @@ public function __construct(ResourceConnection $resourceConnection)
     public function getData(int $stockId, array $sourceIds = []): \ArrayIterator
     {
         $connection = $this->resourceConnection->getConnection();
+        $sourceTable = $connection->getTableName(SourceResourceModel::TABLE_NAME_SOURCE);
         $sourceItemTable = $connection->getTableName(SourceItemResourceModel::TABLE_NAME_SOURCE_ITEM);
         $sourceStockLinkTable = $connection->getTableName(StockSourceLinkResourceModel::TABLE_NAME_STOCK_SOURCE_LINK);
 
+        // find all enabled sources
+        $select = $connection->select()->from($sourceTable, [Source::SOURCE_ID])->where(Source::ENABLED . '=?', 1);
+        if (count($sourceIds)) {
```
Comment (naydav) [inline app/code/Magento/Inventory/Indexer/StockItem/IndexDataProvider.php]: Need to check this condition
If we do not have enabled sources then we need to return empty data

File: app/code/Magento/Inventory/Indexer/StockItem/IndexDataProvider.php [hunk 3]
```diff
@@ -37,9 +39,18 @@ public function __construct(ResourceConnection $resourceConnection)
...
+            $select->where(Source::SOURCE_ID . ' = ?', $sourceIds);
+        }
+        $sourceIds = $connection->fetchCol($select);
```
Comment (naydav) [inline app/code/Magento/Inventory/Indexer/StockItem/IndexDataProvider.php]: Need to check this condition
If we do not have enabled sources then we need to return empty data without extra queries

File: app/code/Magento/InventoryApi/Test/_files/source_items_out_stock.php (hunk 1 above)
Comment (naydav) [reply app/code/Magento/InventoryApi/Test/_files/source_items_out_stock.php]: Need to simplify fixture files
https://github.com/magento-engcom/magento2/pull/76/files#diff-c8b20cde2aec460219d830973d4b6321
like in
https://github.com/magento-engcom/magento2/blob/develop/app/code/Magento/InventoryApi/Test/_files/source_items.php

Now it is possible after merge
https://github.com/magento-engcom/magento2/pull/79
We try to describe more details in our article
https://github.com/magento-engcom/magento2/wiki/The-first-step-towards-pre-generated-IDs.-And-how-this-will-improve-your-Integration-tests

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento-engcom/magento2?pullRequest=76) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-engcom/magento2?pullRequest=76) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: larsroettig<br/>:x: naydav

**Pull request merged**